### PR TITLE
Warn the user when DNSSEC related things are attempted when DNSSEC is disabled

### LINKED ISSUE #4390: Recursor accepts addDS() with dnssec=off
Recursor 4.0.2 accepts `addDS()` from the Lua configuration while `dnssec=off`. 
It would be nice if we could have a warning for this.

### PR DESCRIPTION
**Pull request by pieterlexis**
Fixes #4390

### CODE REVIEW

**Comment by rgacogne** [comment]
Perhaps you could also warn when `addNTA()` is called with DNSSEC disabled?

**Comment by pieterlexis** [comment]
added!

File: pdns/rec-lua-conf.cc [hunk 1]
```diff
@@ -233,26 +234,34 @@ void loadRecursorLuaConfig(const std::string& fname)
 		    });
 
   Lua.writeFunction("addDS", [&lci](const std::string& who, const std::string& what) {
+      if(g_dnssecmode == DNSSECMode::Off)
+        theL()<<Logger::Warning<<"Warning: adding Trust Anchor for DNSSEC (addDS), but dnssec is set to 'off'!"<<endl;
```
Comment (Habbie) [inline pdns/rec-lua-conf.cc]: I see this code duplicated 3 more times with minor edits, maybe make a function?

File: pdns/rec_channel_rec.cc [hunk 2]
```diff
@@ -348,6 +348,9 @@ string doSetCarbonServer(T begin, T end)
 template<typename T>
 string doSetDnssecLogBogus(T begin, T end)
 {
+  if(g_dnssecmode == DNSSECMode::Off)
+    return "DNSSEC is disabled in the configuration, not changing the Bogus logging setting\n";
```
Comment (Habbie) [inline pdns/rec_channel_rec.cc]: more code duplication

**Pull request merged**